src/main/java/com/yahoo/bullet/typesystem/TypedObject.java

```diff
@@ -256,6 +257,39 @@ public int compareTo(TypedObject other) {
         }
     }
 
+    /**
+     * Returns a {@link Comparator} for {@link TypedObject} that, unlike {@link TypedObject#compareTo(TypedObject)},
+     * will work on any one TypedObject being {@link TypedObject#NULL}. In particular, this comparator will place nulls
+     * first when sorting by it. If the first object is {@link TypedObject#NULL}, the comparator will return
+     * {@link Integer#MIN_VALUE}. If the second object is {@link TypedObject#NULL}, the comparator will return
+     * {@link Integer#MIN_VALUE}. Otherwise, the standard {@link TypedObject#compareTo(TypedObject)} will be invoked.
+     *
+     * Note this comparator only works on non-null objects, i.e. the {@link TypedObject} arguments must not be null.
+     *
+     * @return A {@link Comparator} that can places {@link TypedObject#NULL} objects first.
+     */
+    public static Comparator<TypedObject> nullsFirst() {
+        return (o1, o2) -> {
+            boolean firstNull = o1.isNull();
+            boolean secondNull = o2.isNull();
+            // Both null or both non-null
+            if (firstNull == secondNull) {
+                return o1.compareTo(o2);
+            }
+            return firstNull ? Integer.MIN_VALUE : Integer.MAX_VALUE;
+        };
+    }
+
+    /**
+     * This is simply the {@link Comparator#reversed()} on the {@link #nullsFirst()} {@link Comparator}. See docs for
+     * that method. This method places {@link TypedObject#NULL} objects last.
+     *
+     * @return A {@link Comparator} that can places {@link TypedObject#NULL} objects last.
+     */
+    public static Comparator<TypedObject> nullsLast() {
+        return nullsFirst().reversed();
```

Comment:
This is technically not nulls last since the sort order is also reversed. it's nullsfirst but now "sorted descending"

Reply:
Oh whoops dunno why I did nullsLast as desc...